### ISSUE TITLE
Mission with MPC_ALT_MODE terrain follow doesn't work properly

### ISSUE THREAD

**Issue by tuloski** [PX4/Firmware#11266]
I'm testing a mission with terrain_follow enabled with distance sensor, but I think it's not supported.

First of all there is no way to set a waypoint with relative altitude to the terrain. I can only relative to home (in local ned basically) and absolute WGS84.

The problem is that if I use the relative altitude and I have a terrain altitude different from zero, the copter will navigate following the terrain, but it will never reach the desired altitude w.r.t. home hence it will never trigger the waypoint reached in the acceptance radius.

Example home alt is 500m. Target altitude is 20m relative. Terrain altitude at target is 10m. What will happen is that the desired altitude is 520 meters but the copter will reach 530 meters (20 meters from terrain), so there is an error of 10 meters (the terrain altitude) and it will never finish the waypoint.
The altitude setpoint in mission should be modified with the terrain altitude IMO.

**Comment by julianoes** [comment]
@tuloski sorry for the delay. Is this still an issue?

**Comment by tuloski** [comment]
Yes I think it is still an issue but it's months I didn't test it.

**Comment by julianoes** [comment]
Would be good if you could test this again and show a log for us to understand.
